bemani/backend/iidx/rootage.py

```diff
@@ -0,0 +1,1299 @@
+# vim: set fileencoding=utf-8
+import copy
+import struct
+from typing import Optional, Dict, Any
+
+from bemani.backend.iidx.base import IIDXBase
+from bemani.backend.iidx.sinobuz import IIDXSinobuz
+from bemani.backend.iidx.cannonballers import IIDXCannonBallers
+
+from bemani.common import ValidatedDict, Model, VersionConstants, Time, ID, intish
+from bemani.data import Data, UserID
+from bemani.protocol import Node
+
+# todo: clear '?' mark
+class IIDXRootage(IIDXBase):
+
+    name = 'Beatmania IIDX ROOTAGE'
+    version = VersionConstants.IIDX_ROOTAGE
+    sinobuz = None
+
+    # todo: concurrency?
+    def s(self) -> IIDXSinobuz:
+        if self.sinobuz is None:
+            self.sinobuz = IIDXSinobuz(self.data, self.config, self.model)
+        return self.sinobuz
+
+
+    # proxied: sinobuz.py

```

Comment:
Prefer copypasta over proxy, this was intentionally chosen to bias towards ability to change individual games without unintentially breaking forwards or backwards compatibility with other games. Its fine to have identical code, since the opposite end is something we don't control and the code is better documenting as to what's going on in that particular game version.